_posts/2018-12-17-the-lazy-applicative-functor.html

```diff
@@ -21,7 +21,7 @@ <h3 id="c04ca371d39847cc86a90240a7f53745">
 		As you have <a href="/2018/10/15/an-applicative-password-list">previously seen</a>, C# isn't the best fit for the concept of applicative functors. Nevertheless, you can write an <code>Apply</code> extension method following the applicative 'code template':
 	</p>
 	<p>
-		<pre><span style="color:blue;">public</span>&nbsp;<span style="color:blue;">static</span>&nbsp;<span style="color:#2b91af;">Lazy</span>&lt;<span style="color:#2b91af;">TResult</span>&gt;&nbsp;Apply&lt;<span style="color:#2b91af;">TResult</span>,&nbsp;<span style="color:#2b91af;">T</span>&gt;(
+		<pre><span style="color:blue;">public</span>&nbsp;<span style="color:blue;">static</span>&nbsp;<span style="color:#2b91af;">Lazy</span>&lt;<span style="color:#2b91af;">TResult</span>&gt;&nbsp;Apply&lt;<span style="color:#2b91af;">T</span>,&nbsp;<span style="color:#2b91af;">TResult</span>&gt;(
```

Comment:
Why do you want to flip the type arguments?

FWIW, this code snippet is copied from code that compiles and has passing tests.

Reply:
Oh, I see, sorry. It was kind of a reflex, after reading several applicative-related posts in the series (like the password list one, the maybe one and the test data one) where the order is always `<T, TResult>`.

That difference kind of emerged and I thought there was some kind of convention, as in Either's left and right. My wrong. If it's ok for you I'll revert this change then.

Reply:
It may be inconsistent, but I prefer not to edit code in published blog posts if I can avoid it.

I already don't trust myself with editing real code without a type system and test coverage.

I _could_ edit the code in the underlying code base and then copy and paste the edit into the blog post, but that then entails that I may have to update all the other places I've included or used that piece of code, including other articles... That's quite an undertaking.